lib/nats.js

```diff
@@ -1905,6 +1909,40 @@ Client.prototype.requestOne = function(subject, opt_msg, opt_options, timeout, c
     return this.request(subject, opt_msg, opt_options, callback);
 };
 
+// finds the last index that satisfies the testFun in array a
+function lastIndexOf(a, testFun) {
+    const idx = a.length-1;
+    for(let i=idx; i > -1; i--) {
+        if (testFun(a[i])) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+// returns an array that shifts the last function in the
+// array to the last element of the array
+function callbackShifter(maxArgs, args) {
+    // last argument is expected to be a function
+    const a = new Array(maxArgs);
+    const idx = lastIndexOf(args, (v) => typeof v === 'function');
+
+    // found a function, at the last index
+    if (idx !== -1) {
+        a[maxArgs - 1] = args[idx];
```

Comment:
Looks to me that you would lose the last element if that is not a function, and you would still keep the function at its original position. Suppose you have:
`1, 2, 3, F, 4, 5` where `F` is a function. The `lastIndexOf()` would return idx 3. In array `a` you would put `a[5]='F'`, and then copy only from 0 to 2, so `a` would result in: `1, 2, 3, _,  _, F` if I understand your code.. (here the `_` means that there is no value copied from `args`, so undefined I guess.

Reply:
the callbacks always end in a function, so arguments 4, 5 and not valid, and are dropped - this is is correct.

Reply:
but maxArgs is 6 in the above case, no?